### PR TITLE
Fixed the issue for the clearqueryparams 

### PR DESCRIPTION
**Pull request by Arti-Khillare1**

### CODE REVIEW

File: projects/self-creation-portal/src/app/services/common-service/common.service.ts [hunk 1]
```diff
@@ -53,7 +53,17 @@ export class CommonService {
 
   clearQueryParams() {
     const queryParams = this.route.snapshot.queryParams;
-    if (Object.keys(queryParams).length > 0) {
+    const currentPath = this.router.url.split('?')[0];
+    const sidenavPaths = [
+      '/home/drafts',
+      '/home/up-for-review',
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/common-service/common.service.ts]: Why hard coded routes ?

File: projects/self-creation-portal/src/app/services/common-service/common.service.ts [hunk 2]
```diff
@@ -53,7 +53,17 @@ export class CommonService {
...
+      '/home/submitted-for-review'
+    ];
+    // Check if the current path matches any of the tab paths, if matches then clear params and redirect to route
+    const issidenavPaths = sidenavPaths.some(tabPath => currentPath.startsWith(tabPath));
+  
+    if ((currentPath.includes('solution') && issidenavPaths && Object.keys(queryParams).length > 0) ||
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/common-service/common.service.ts]: Can you explain "solution" string comparision

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/self-creation-portal/src/app/services/common-service/common.service.ts (hunk 2 above)
Comment (Arti-Khillare1) [reply projects/self-creation-portal/src/app/services/common-service/common.service.ts]: as it is related to this url "/solution/project/project-details" then it should not clear the params and getback to original route.
when searching after some searchtext and clicking on the button then ngDestroy detecting it coming from different component so to handle that case I have tried here only when this particular sidenavpaths are there then only clearparams else don't clear, so to fixed this issue I am checking currentpath includes solutions url.

Comment (ramkumar-ns-dev) [reply projects/self-creation-portal/src/app/services/common-service/common.service.ts]: The please define as parent route and compare here, don't hard code

File: projects/self-creation-portal/src/app/services/common-service/common.service.ts [hunk 3]
```diff
@@ -53,7 +53,23 @@ export class CommonService {
...
   clearQueryParams() {
     const queryParams = this.route.snapshot.queryParams;
-    if (Object.keys(queryParams).length > 0) {
+    const currentPath = this.router.url.split('?')[0];
+    const parentRoutes = this.route.parent?.snapshot.url.map(segment => segment.path).join('/') || '';
+    const childRoutes = this.route.routeConfig?.children?.map(route => `${parentRoutes}/${route.path}`) || [];
+    
+    // Filter relevant sidenav paths that match the routes
+    const sidenavPaths = childRoutes.filter(routePath => 
+      ['drafts', 'up-for-review', 'submitted-for-review'].some(sidenav => routePath.includes(sidenav))
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/common-service/common.service.ts]: As I already mentioned don't use string comparison of project status, use constant variables

File: projects/self-creation-portal/src/app/services/common-service/common.service.ts [hunk 4]
```diff
@@ -53,7 +53,23 @@ export class CommonService {
...
+    );
+
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/common-service/common.service.ts]: Why route hard coded ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**